### PR TITLE
Historian fixes for issues 1314, 1315, 1316, 1317, and 1346

### PR DESCRIPTION
**Pull request by schandrika**
Historian fixes for issues 
#1314 
#1315  
#1316   
#1317
#1346 

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1326)

### CODE REVIEW

**Comment by craig8** [comment]
Why is every single one of the test not passing?

**Comment by jhaack** [comment]
[1m[31mvolttrontesting/utils/platformwrapper.py[0m:493: NameError
 ERROR at setup of test_correct_admin_permissions_on_vcp_vc_and_listener_agent _

Name errors on these failing tests..

**Comment by schandrika** [comment]
That error is due to an earlier pull request merge into develop.  I will check and fix it.

**Comment by schandrika** [comment]
Pulled from upstream and merged again

File: services/core/MongodbHistorian/mongodb/historian.py [hunk 1]
```diff
@@ -660,36 +645,49 @@ def query_historian(self, topic, start=None, end=None, agg_type=None,
...
+
+        if ts_filter:
+            if start == end:
+                find_params = {'ts': start}
+            else:
+                find_params = {'ts': ts_filter}
+
         find_params['topic_id'] = ObjectId(topic_id)
-        _log.debug("querying table with params {}".format(find_params))
+        _log.debug("{}:Querying topic {}".format(topic_id, topic_id))
```
Comment (craig8) [inline services/core/MongodbHistorian/mongodb/historian.py]: not sure why topic_id topic_id for the params to format

**Comment by hlngo** [comment]
Do we have ETA for this PR?

**Comment by schandrika** [comment]
Merged changes from develop. Added tagging service specification that got accidentally deleted earlier

File: docs/source/specifications/tagging_service.rst [hunk 2]
```diff
@@ -0,0 +1,350 @@
+=============================
+Tagging agent specification
```
Comment (craig8) [inline docs/source/specifications/tagging_service.rst]: Please only add enough === to the end of the title

**Comment by hlngo** [comment]
No progress on this PR?

**Pull request merged**